inc/widgets/odm-contents-same-category-widget.php

```diff
@@ -39,15 +38,24 @@ public function widget( $args, $instance ) {
 
 			//TODO: OPtimize this query to filter out categories directly
 			// and ensuring $limit is precise.
-
+			$filter_by_lang = strtolower(get_the_language_by_language_code(odm_language_manager()->get_current_language()));
 			$query = array(
 					'posts_per_page'   => $limit,
 					'order'            => 'DESC',
 					'post_type'        => $supported_post_types,
-					'post_status'      => 'publish'
+					'category' 				 => $categories,
+					'post_status'      => 'publish'/*,
+					'tax_query'				 => array(
```

Comment:
Well,  the filter the post by category is fixed now with adding "'category' =>$categories,". The comment codes just would like to filter the post if the languages is set. the comment code is correct to filter those posts if language is set, but it is not working right now,  so i just comment those code for now, and will continue to fix it later. You can remove it if you want.